src/seedu/addressbook/data/person/Name.java

```diff
@@ -43,21 +43,31 @@ public int hashCode() {
         return fullName.hashCode();
     }
     
-    public boolean isSimilar(Name other) {
+    public boolean isSimilar_notNull(Name other) {
     	if (other == null) {
     		return false;
     	} else {
-    		if (fullName.equalsIgnoreCase(other.toString())) {
-    			return true;
-    		} else {
-    			List<String> splitOther = other.getWordsInName();
-    			for (String string : splitOther) {
-    				if (fullName.contains(string)) {
-    					return true;
-    				}
-    			}
-    			return false;
+    		return true;
     		}
+    }
+    	
+    public boolean isSimilar_caseSensitive(Name other) {
+    	return (fullName.equalsIgnoreCase(other.toString()));
+    }
+    
+    public boolean isSimilar_subset(Name other) {
+    	String[] splitSubSetOther = other.toString().split(" ");
+    	for (String string : splitSubSetOther) {
+    		return (fullName.contains(string));
+    	}
+    	return false;
+    }
+    
+    public boolean isSimilar_differentOrderWithComma(Name other) {
+    	List<String> splitOther = other.getWordsInName();
+    	for (String string : splitOther) {
+    		return (fullName.contains(string));
     	}
+    	return false;
     }
```

Comment:
Please fix the indentation and change the tab to spaces in Eclipse, see [here](http://stackoverflow.com/questions/407929/how-do-i-change-eclipse-to-use-spaces-instead-of-tabs)